spark/src/test/java/org/apache/zeppelin/spark/IPySparkInterpreterTest.java

```diff
@@ -0,0 +1,203 @@
+      assertEquals(InterpreterResult.Code.SUCCESS, result.code());
+      interpreterResultMessages = context.out.getInterpreterResultMessages();
+      assertEquals(
+          "+---+---+\n" +
+          "| _1| _2|\n" +
+          "+---+---+\n" +
+          "|  1|  a|\n" +
+          "|  2|  b|\n" +
+          "+---+---+\n\n", interpreterResultMessages.get(0).getData());
+    }
+
+    // cancel
+    final InterpreterContext context2 = getInterpreterContext();
+
+    Thread thread = new Thread(){
+      @Override
+      public void run() {
+        InterpreterResult result = iPySparkInterpreter.interpret("import time\nsc.range(1,10).foreach(lambda x: time.sleep(1))", context2);
+        assertEquals(InterpreterResult.Code.ERROR, result.code());
+        List<InterpreterResultMessage> interpreterResultMessages = null;
+        try {
+          interpreterResultMessages = context2.out.getInterpreterResultMessages();
+          assertTrue(interpreterResultMessages.get(0).getData().contains("cancelled"));
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    };
+    thread.start();
+
+    // sleep 1 second to wait for the spark job starts
+    Thread.sleep(1000);
+    iPySparkInterpreter.cancel(context);
+    thread.join();
+
+    // completions
+    List<InterpreterCompletion> completions = iPySparkInterpreter.completion("sc.ran", 6, getInterpreterContext());
+    assertEquals(1, completions.size());
+    assertEquals("sc.range", completions.get(0).getValue());
+
+    // pyspark streaming
+    context = getInterpreterContext();
+    result = iPySparkInterpreter.interpret(
+        "from pyspark.streaming import StreamingContext\n" +
+        "import time\n" +
+        "ssc = StreamingContext(sc, 1)\n" +
+        "rddQueue = []\n" +
+        "for i in range(5):\n" +
+        "    rddQueue += [ssc.sparkContext.parallelize([j for j in range(1, 1001)], 10)]\n" +
+        "inputStream = ssc.queueStream(rddQueue)\n" +
+        "mappedStream = inputStream.map(lambda x: (x % 10, 1))\n" +
+        "reducedStream = mappedStream.reduceByKey(lambda a, b: a + b)\n" +
+        "reducedStream.pprint()\n" +
+        "ssc.start()\n" +
+        "time.sleep(6)\n" +
+        "ssc.stop(stopSparkContext=False, stopGraceFully=True)", context);
+    Thread.sleep(100);
+    assertEquals(InterpreterResult.Code.SUCCESS, result.code());
+    interpreterResultMessages = context.out.getInterpreterResultMessages();
+    assertEquals(1, interpreterResultMessages.size());
+    assertTrue(interpreterResultMessages.get(0).getData().contains("(0, 100)"));
+  }
+
+  private InterpreterContext getInterpreterContext() {
+    return new InterpreterContext(
+        "noteId",
+        "paragraphId",
+        "replName",
+        "paragraphTitle",
+        "paragraphText",
+        new AuthenticationInfo(),
+        new HashMap<String, Object>(),
+        new GUI(),
+        null,
+        null,
+        null,
+        new InterpreterOutput(null));
```

Comment:
IPySparkInterpreterTest will run all the test of IPythonInterpreterTest. see L94